src/axom/sidre/examples/spio/IOSCRCkpt.cpp

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**************************************************************************
+ *************************************************************************/
+
+#include "mpi.h"
+
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/sidre.hpp"
+
+#ifdef AXOM_USE_SCR
+#include "scr.h"
+#endif
+
+using axom::sidre::Group;
+using axom::sidre::DataStore;
+using axom::sidre::DataType;
+using axom::sidre::IOManager;
+using namespace axom::utilities;
+
+/**************************************************************************
+ * Subroutine:  main
+ * Purpose   :
+ *************************************************************************/
+
+int main(int argc, char* argv[])
+{
+  MPI_Init(&argc, &argv);
```

Comment:
@adammoody -- can you please refactor this example to have the following structure, with `main` as a skeleton and all the actual SCR work implemented in functions?

```cpp
int main()
{
  // init MPI and SCR

  // parse command line args

  // The data store should exist before attempting to restart
  DataStore ds = new Datastore(...);

  // attempt to load restart; all SCR calls for loading restart here
  bool hasRestart = loadRestart(file_base, num_ckpts, ds); 
   if( !hasRestart )
   {
       initializeDataStore(ds);   // if not restart, initialize groups and views as desired
   }

   //  do some work
   forall checkpoints 
   {
        // update state in existing datastore
        //  i.e. don't create views or groups here

       // attempt to create checkpoint; all SCR calls for dumping checkpoint here
        dumpCheckpoint(ds);
   }

   // finalize SCR and MPI
}
```

Reply:
@adammoody -- this is the main change that I'm requesting for this PR.  The others were mainly suggestions about possible improvements.

Reply:
I've updated the IO_SCR_Ckpt example for this.  I haven't created functions for the datastore, but I reduced each of those to a few lines each so that it's simpler.  Is that reduced datastore code simple enough as it is, or would you still like to have functions here?

There are a few spots where it comes up:
- reading values from the datastore into local variables after reading a restart
- initializing the datastore if there was no restart to read
- updating datastore values before writing a checkpoint